fuel-storage/src/lib.rs

```diff
@@ -123,6 +123,14 @@ pub struct StorageRef<'a, T: 'a + ?Sized, Type: Mappable>(&'a T, core::marker::P
 pub trait StorageAsRef {
     #[inline(always)]
     fn storage<Type>(&self) -> StorageRef<Self, Type>
+    where
+        Type: Mappable,
```

Comment:
What is the main motivation behind this PR? Is it blocking you somewhere else? If so do we still have time to sneak it in before next release for everything?

Reply:
It it to improve some part of the code in the future to avoid `(&mut database)` or `(&**database)`